src/pudl_rmi/coordinate.py

```diff
@@ -336,6 +411,120 @@ def check_is_file_or_not_exists(file_path: Path):
         )
 
 
+def pickle_plant_parts_distinct(plant_parts_eia, file_path):
+    """Pickle dataframe with only unique granularities from PPE."""
+    # Figure out where this should go. It's currently duplicated in
+    # connect_ferc1_to_eia.InputManager, should it be a PUDL function?
+    # can this be done more efficiently? use pipe?
+    # Anyways,
+    # We want only the records of the EIA plant-parts that are "true
+    # granularies" and those which are not duplicates based on their
+    # ownership  so the model doesn't get confused as to which option to
+    # pick if there are many records with duplicate data
+    plant_parts_eia = plant_parts_eia.assign(
+        plant_id_report_year_util_id=lambda x: x.plant_id_report_year
+        + "_"
+        + x.utility_id_pudl.map(str)
+    ).astype({"installation_year": "float"})
+    plant_parts_distinct = plant_parts_eia[
+        (plant_parts_eia["true_gran"]) & (~plant_parts_eia["ownership_dupe"])
+    ]
+    plant_parts_distinct.to_pickle(file_path)
+
+
+def _set_plant_part_dtypes(plant_parts_eia):
+    """Set column data types to reduce memory usage."""
+    plant_parts_eia.index = plant_parts_eia.index.astype("string")
+    plant_parts_eia = plant_parts_eia.astype(
+        {
+            "plant_part_id_eia": "string",
+            "appro_record_id_eia": "string",
+            "technology_description": "category",
+            "energy_source_code_1": "category",
+            "plant_part": "category",
+            "plant_id_report_year": "string",
+            "operational_status_pudl": "category",
+            "appro_part_label": "category",
+            "ownership": "category",
+            "ferc_acct_name": "category",
+            "fuel_type_code_pudl": "category",
+            "prime_mover_code": "category",
+        }
+    )
+    return plant_parts_eia
+
+
+def prep_train_connections(ppe):
```

Comment:
This function now hangs out here as a helper function.

Reply:
I feel like it would be better too keep this in the `connect_fercs1_to_eia` module. Even if it is pulled out of the og `InputManager` class. Mostly because these train connections are only really relevant for that stage of the processing. Even if `prep_train_connections` would only actually be called inside of this module. But idk this could be a silly instinct 🤷🏻

Reply:
I moved `prep_train_connections` back into `connect_ferc_to_eia` module.